samples/Tutorials/CSharpIntermediate/CSharpIntermediate/CSharpIntermediate.Game/Utils/TutorialUI.cs

```diff
@@ -0,0 +1,149 @@
+                { "Async Scripts", "05_Async/Completed-AsyncScript" },
+                { "Scene loading", "06_SceneLoading/Completed-SceneA" },
+                { "Animation basics", "07_Animation-basics/Completed-Animations" },
+                { "Audio", "08_Audio/Completed-Audio" },
+                { "First person camera", "09_FirstPersonCamera/Completed-FirstPersonCamera" },
+                { "Third person camera", "10_ThirdPersonCamera/Completed-ThirdPersonCamera" },
+                { "Navigation", "11_Navigation/Completed-Navigation" }
+            };
+
+            CreateButton(completedButton, complextedText, tutorialScenes, buttonsCompletedUI);
+
+            buttonsStartUI.Children.Remove(startButton);
+            buttonsCompletedUI.Children.Remove(completedButton);
+        }
+
+        private void CreateButton(Button baseButtonButton, TextBlock textBlock, Dictionary<string, string> tutorialScenes, StackPanel stackPanel)
+        {
+            foreach (var keyPair in tutorialScenes)
+            {
+                var button = new Button
+                {
+
+                    Content = new TextBlock
+                    {
+                        Font = textBlock.Font,
+                        TextSize = textBlock.TextSize,
+                        Height = baseButtonButton.Content.Height,
+                        Text = keyPair.Key,
+                        TextColor = Color.White,
+                        HorizontalAlignment = HorizontalAlignment.Left,
+                        VerticalAlignment = VerticalAlignment.Center
+                    },
+                    Height = baseButtonButton.Height,
+                    NotPressedImage = baseButtonButton.NotPressedImage,
+                    PressedImage = baseButtonButton.PressedImage,
+                    MouseOverImage = baseButtonButton.MouseOverImage,
+
+                };
+                button.Click += (sender, e) => BtnLoadTutorial(sender, e, keyPair);
+
+                stackPanel.Children.Add(button);
+            }
+        }
+
+        private void BtnTutorialMenuClicked(object sender, RoutedEventArgs e)
+        {
+            tutorialMenu.Visibility = tutorialMenu.IsVisible ? Visibility.Hidden : Visibility.Visible;
+            buttonsStartUI.Visibility = buttonsStartUI.IsVisible ? Visibility.Hidden : Visibility.Visible;
+            buttonsCompletedUI.Visibility = buttonsCompletedUI.IsVisible ? Visibility.Hidden : Visibility.Visible;
+
+            if (tutorialMenu.Visibility == Visibility.Visible)
+            {
+                Game.Window.IsMouseVisible = true;
+            }
+        }
+
+        private void BtnLoadTutorial(object sender, RoutedEventArgs e, KeyValuePair<string, string> newTutorialScene)
+        {
+            if (SceneSystem.SceneInstance.RootScene.Children.Count > 0)
+            {
+                SceneSystem.SceneInstance.RootScene.Children.RemoveAt(0);
```

Comment:
Should you unload the child scene here? Or if not, then below it might be worth to first try `Content.Get<Scene>` so that we don't increase ref counter with each `Load`, not that it matters too much here...